packages/expo-cli/src/credentials/views/AndroidKeystore.ts

```diff
@@ -0,0 +1,231 @@
+import os from 'os';
+import path from 'path';
+
+import { AndroidCredentials } from '@expo/xdl';
+import chalk from 'chalk';
+import omit from 'lodash/omit';
+import fs from 'fs-extra';
+import ora from 'ora';
+import { v4 as uuid } from 'uuid';
+
+import { askForUserProvided, getCredentialsFromUser } from '../actions/promptForCredentials';
+import { Context, IView } from '../context';
+import { keystoreSchema } from '../credentials';
+import prompt, { Question } from '../../prompt';
+import log from '../../log';
+
+type Keystore = AndroidCredentials.Keystore;
+
+class UpdateKeystore implements IView {
+  constructor(private experienceName: string) {}
+
+  async open(ctx: Context): Promise<IView | null> {
+    const keystore = await this.provideOrGenerate(ctx);
+
+    await ctx.android.updateKeystore(this.experienceName, keystore);
+    log(chalk.green('Keystore updated successfully'));
+    return null;
+  }
+
+  async provideOrGenerate(ctx: Context): Promise<Keystore> {
+    const providedKeystore = await askForUserProvided(keystoreSchema);
+    if (providedKeystore) {
+      return providedKeystore;
+    }
+
+    const tmpKeystoreName = path.join(
+      os.tmpdir(),
+      `${this.experienceName}_${uuid()}_tmp.jks`.replace('/', '__')
+    );
+    try {
+      await fs.remove(tmpKeystoreName);
+      const keystoreData = await AndroidCredentials.generateUploadKeystore(
+        tmpKeystoreName,
+        '--------------',
+        this.experienceName
+      );
+
+      return {
+        ...omit(keystoreData, 'keystorePath'),
+        keystore: await fs.readFile(tmpKeystoreName, 'base64'),
+      };
+    } catch (error) {
+      log.warn(
+        "If you don't provide your own Android keystore, it will be generated on our servers during the next build"
```

Comment:
This message is a little weird. If I understand the code in `try` correctly - the error can be thrown if we failed to generate a keystore. For example, if the `keytool` command had failed.